compiler/il/OMRDataTypes.cpp

```diff
@@ -158,6 +158,15 @@ OMR::DataType::getSize(TR::DataType dt)
             TR_ASSERT_FATAL(false, "Incorrect Vector Length\n");
          }
       }
+   else if (dt.isMask())
+      {
```

Comment:
On x86, different microarchitectures (with AVX-512) would have a different internal representation in the mask registers. I'm concerned that this would cause the mask types to have different names, depending on the target microarchitecture. I'm guess that would cause issues in TRIL and confusion when reading the logs. Are there any other reasons to need the mask type size?

Reply:
Every type needs to have a size, for example to know the size of temps to be allocated. But this leads to some interesting questions. When we print Vector and Mask type names we print corresponding vector lengths and element types. For example, if we deal with Float128Vector Vector API class, vector type will be printed as Vector128Float and mask type will be printed as Mask128Float.  But the actual, physical, mask size on AVX-512 is just 64-bits. So we need to distinguish "vector length/num lanes" the mask corresponds to from its physical length. Need to think about it a bit...

Reply:
I think we found a good solution in the last commit.